picard/config.py

```diff
@@ -129,6 +131,53 @@ def value(self, name, option_type, default=None):
         return default
 
 
+class SettingConfigSection(ConfigSection):
+    """Custom subclass to automatically accommodate saving and retrieving values based on user profile settings.
+    """
+    def __init__(self, config, name):
+        super().__init__(config, name)
+        self.__qt_config = config
+        self.__name = name
+        self.__prefix = self.__name + '/'
+        self._memoization = defaultdict(Memovar)
+        Option.add_if_missing(name, 'user_profiles', [])
+
+    def _get_active_profiles(self):
+        profiles = self._get_profiles()
+        for profile in profiles:
+            if profile['active']:
+                yield profile
+
+    def _get_profiles(self):
+        profiles = self['user_profiles']
+        return profiles if profiles is not None else []
+
+    def __getitem__(self, name):
+        # Don't process settings that are not profile-specific
+        if name in UserProfileGroups.get_all_settings_list():
+            for profile in self._get_active_profiles():
+                if name in profile['settings'] and profile['settings'][name] is not None:
+                    return profile['settings'][name]
+        opt = Option.get(self.__name, name)
+        if opt is None:
+            return None
+        return self.value(name, opt, opt.default)
+
+    def __setitem__(self, name, value):
+        # Don't process settings that are not profile-specific
+        if name in UserProfileGroups.get_all_settings_list():
+            profiles = self._get_profiles()
```

Comment:
We can use `_get_active_profiles()` here too I think.

Reply:
I'm not sure that we can because we need the full list of profiles to write back to the "user_profiles" setting.  I'll do some testing just to make sure.

An alternative **might** be to store the list as an ordered dictionary, or as a dictionary with some sort of sort key.

Reply:
> **EDIT 2:** Another alternative would be to store the actual settings for each profile in a dictionary keyed on unique profile ID (uuid?) with the id, activated status and title in a list of tuples (ordered by position in the profile stack). That may provide faster access to the settings, and avoid having to re-write the whole list every time a setting is updated.

Yes, this approach is likely better, worth a try imho.